src/transformers/generation/configuration_utils.py

```diff
@@ -608,7 +632,13 @@ def validate(self, is_init=False):
                     f"({self.num_beams})."
                 )
 
-        # 5. check common issue: passing `generate` arguments inside the generation config
+        # 5. check `cache_config`
+        if self.cache_config is not None:
+            if not isinstance(self.cache_config, QuantizedCacheConfig):
```

Comment:
We can probably have a mapping from cache implementation to cache config, this is not really scalable!

Reply:
You mean hardcoding the `QuantizedCacheConfig` class? Yes, this works only for the current class, since we do not have any more configs yet, 

Oke, I can make a mapping similar to what we have in `generate()` for cache_implementations

Reply:
sounds good! 🤗